src/subnet.c

```diff
@@ -86,6 +147,32 @@ void free_subnet(subnet_t *subnet) {
 	free(subnet);
 }
 
+void subnet_cache_flush(subnet_t *subnet) {
+	switch(subnet->type) {
+	case SUBNET_IPV4:
+		if(subnet->net.ipv4.prefixlength == 32) {
+			hash_delete(ipv4_t, &ipv4_cache, &subnet->net.ipv4);
```

Comment:
The third argument should be `&subnet->net.ipv4.address`. The same goes for the IPv6 and MAC subnets.

Reply:
Agreed.

Reply:
Agreed. Updated.